packages/dds/test-dds-utils/src/ddsSnapshotHarness.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+
+// Simple filter to avoid tests with a name that would accidentally be parsed as directory traversal or other confusing things.
+const nameCheck = new RegExp(/^[^"/\\]+$/);
+
+let currentTestName: string | undefined;
+let currentTestFile: string | undefined;
+
+/**
+ * A utility function for setting up a snapshot directory for tests.
+ * @param snapshotsFolder - The folder where snapshots are stored.
+ * @param dirPath - The directory within the snapshots folder where tests will store their snapshots with specific
+ * catogories. Defaults to root directory ("/").
+ * @internal
+ */
+export function useSnapshotDirectory(snapshotsFolder: string, dirPath: string = "/"): void {
+	assert(existsSync(snapshotsFolder));
+
+	const normalizedDir = path.join(snapshotsFolder, dirPath);
+	// Basic sanity check to avoid bugs like accidentally recursively deleting everything under `/` if something went wrong (like dirPath navigated up directories a lot).
+	assert(normalizedDir.startsWith(snapshotsFolder));
+
+	beforeEach(function (): void {
+		currentTestName = this.currentTest?.title ?? assert.fail();
+		currentTestFile = path.join(normalizedDir, `${currentTestName}.json`);
+	});
+
+	afterEach(() => {
+		currentTestFile = undefined;
+		currentTestName = undefined;
+	});
+}
+
+/**
+ * @internal
+ */
+export interface TestScenario {
```

Comment:
this should stay in the directory fuzz tests until we're confident something like it is the right abstraction (and even then, it wouldn't go in this file IMO)